projects/core/components/data-list/data-list.component.ts

```diff
@@ -86,6 +88,7 @@ export class TuiDataListComponent<T>
     public ngAfterContentChecked(): void {
         // TODO: Refactor to :has after Safari support bumped to 15
         this.empty = !this.el.querySelector('[tuiOption]');
+        this.cdr.markForCheck();
```

Comment:
Maybe it's better to do `detectChanges`? So that only this view is checked, also we should only do it if `empty` value changed.